diff --git a/dlls/wow64/syscall.c b/dlls/wow64/syscall.c
index 4720fe134fad9fd6a4e9e8634a5e0bc968ce35a3..21511c32ac4fca89f5fe72f1c183472a2b9591a2 100644
--- a/dlls/wow64/syscall.c
+++ b/dlls/wow64/syscall.c
@@ -572 +572,9 @@ NTSTATUS WINAPI wow64_NtGetContextThread( UINT *args )
-    if (wow64info->CpuFlags & WOW64_CPUFLAGS_SOFTWARE) RtlWow64SuspendThread( handle, &count );
+    THREAD_BASIC_INFORMATION info;
+    BOOL self = (handle == GetCurrentThread());
+
+    if (!self && !NtQueryInformationThread( handle, ThreadBasicInformation,
+                                             &info, sizeof(info), NULL ))
+        self = HandleToULong( info.ClientId.UniqueThread ) == GetCurrentThreadId();
+
+    if ((wow64info->CpuFlags & WOW64_CPUFLAGS_SOFTWARE) && !self)
+        RtlWow64SuspendThread( handle, &count );
@@ -574 +582,2 @@ NTSTATUS WINAPI wow64_NtGetContextThread( UINT *args )
-    if (wow64info->CpuFlags & WOW64_CPUFLAGS_SOFTWARE) NtResumeThread( handle, &count );
+    if ((wow64info->CpuFlags & WOW64_CPUFLAGS_SOFTWARE) && !self)
+        NtResumeThread( handle, &count );
